### DISCUSSION TITLE
Internationalized routing + custom _app.js, where is "locale" variable?

### DISCUSSION THREAD

**Discussion by ProjectINT** [vercel/next.js#18668] [Help]
The problem is in that i require locale variable in _app.js but can't find it.
In my custom app i have next render:
`
render(): React$Element<typeof AppContextProvider> {
    const {
      Component, pageProps, headerProps, footerProps,
    } = this.props;

    return (
      <AppContextProvider>
        <Head>
          <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
        </Head>
          <Header {...headerProps} />
          <Component {...pageProps} />
          <Footer {...footerProps} />
...
`
in footer and header i have getStaticProps, becouse i want to optimize it to avoid reload this parts of pages, when page is changed.
But in header and footer getStaticProps have not locale variable.
In _app.js the same, const { locale } = context; // undefined
Please help to decide how to resolve this problem with use new 10 version  Internationalized routing feature.
Are this possible?
May be i'am doing something wrong?

**Comment by w01fgang** [comment]
I can confirm, `locale` property is `undefined` in the context inside `_app.js`, but available inside `getStaticProps` exported from any file inside `pages` derectory.
